### DISCUSSION TITLE
Allow using id of state in state.matches

### DISCUSSION THREAD

**Discussion by bartenra** [statelyai/xstate#3135] [Ideas]
For example:

```
state.matches('#Modal.createNavPointForm')
```

Right now I am adding `tags` everywhere to use `modal.hasTag`.

```
createNavPointForm: {
  tags: ['createNavPointForm'],
```

It would be cleaner if I could re-use the same syntax that can be used with `target`s.

**Comment by Andarist** [comment]
I think this is a good idea. Perhaps `.matches` doesn't fit better when we consider partial state values but on the other hand, it's not worth being puristic here. A good, dedicated, name for checking IDs would be something like `.contains` but then... having multiple methods for checking a very similar thing isn't that great.

At the moment this kinda could clash with state keys starting with `#` because we didn't make it 100% clear that it's a bad idea... but I think that's also a puristic concern. Using `#` at the beginning of a state key is quite an obvious code smell.

**Reply by davidkpiano** [reply]
We should make keys that start with `#` invalid, on that note.
